ui/prometheus-plugin/src/plugins/prometheus-time-series-query/get-time-series-data.ts

```diff
@@ -29,6 +29,11 @@ export const getTimeSeriesData: TimeSeriesQueryPlugin<PrometheusTimeSeriesQueryS
   spec,
   context
 ) => {
+  if (spec.query === undefined || spec.query === null || spec.query === '') {
+    // Do not make a request to the backend, instead return an empty TimeSeriesData
+    return { series: [] };
+  }
```

Comment:
I could also move this down and use the timeRange + stepMs that are calculated below, and then we don't need to make `timeRange` and `stepMs` optional... 

Open to thoughts

Reply:
This turned out to be pretty straightforward. 👍 

I think the only downside here is we're not differentiating between "blocked because the query is empty" and "empty results cause there's no data". I wonder if that's an important distinction? When I was thinking through this initially, I was thinking this might be another case like @saminzadeh had with variables where `dependsOn` might come into play.

Maybe this is good enough? Or maybe it's good short-term but we need to differentiate long-term? Not sure.

Reply:
Played around with it a little bit and noticed one weird case where when you multiple queries and one is empty, you can occasionally see 'No data' flash as the loading state, see recording here:

https://user-images.githubusercontent.com/9369625/202294746-b22df68b-c97e-4622-9045-66b79418f94a.mp4

Reply:
My 2 cents: I don't really think there is a need to differentiate between "Empty results" and "You must provide a query". FWIW Grafana does it this way as well and I personally don't mind that experience. 

If we go the `dependsOn` route and use our current implementation, it would basically be stuck in a loading state which is a bad experience IMO. Nice thing is that this is relatively an easy thing to change so maybe we should move forward with this and revisit if the feedback is negative.

Reply:
^ That makes a lot of sense to me. I think we can think about the "1 of the 4 queries" fail use-case in the next iteration.

Reply:
I think we should come back to the loading state too though, if one or multiple queries take a lot longer than others, it was nice how they rendered as each finished, but something we can look into later, this is a lot more clear and fixes the main issue 🙌

Reply:
> I think we should come back to the loading state too though, if one or multiple queries take a lot longer than others, it was nice how they rendered as each finished, but something we can look into later, this is a lot more clear and fixes the main issue

If we do that, getting the UX "right" is going to be crucial. It should be clear that there is more data inbound (almost like streaming results). 

I would caution on the side of showing the data once everything is ready to prevent confusion and simulate a batch request to a backend, almost as if they are one query returning multiple results. But agreed, definitely something to look into.